test/cli/test_login.rb

```diff
@@ -0,0 +1,272 @@
+    error_pattern = /No such record/
+
+    # fail to retrieve nonexistent login:
+    run_command "kbsecret login -x #{label}" do |cmd|
+      cmd.wait
+      assert_match error_pattern, cmd.stderr
+    end
+  end
+
+  def test_no_label
+    label = "test-no-label"
+    username = "user"
+    password = "pass"
+    error_pattern = /Too few arguments given/
+
+    # create login:
+    run_command "kbsecret new login -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # fail to retrieve login without label:
+    run_command "kbsecret login" do |cmd|
+      cmd.wait
+      assert_match error_pattern, cmd.output
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_session
+    session = "test-session"
+    default_label = "test-default-login"
+    session_label = "test-session-login"
+    username = "user"
+    password = "pass"
+    default_output = "#{default_label}:#{username}:#{password}"
+    session_output = "#{session_label}:#{username}:#{password}"
+    error_pattern_no_record = /No such record/
+    error_pattern_no_session = /Unknown session/
+
+    # create default login:
+    run_command "kbsecret new login -x #{default_label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # create a new session
+    run_command_and_stop "kbsecret session new #{session} -r test"
+
+    # create login in session:
+    run_command "kbsecret new login -s #{session} -x #{session_label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve default login:
+    run_command "kbsecret login -s default -x #{default_label}" do |cmd|
+      cmd.wait
+      assert_equal default_output, cmd.output.chomp
+    end
+
+    # retrieve login from session:
+    run_command "kbsecret login -s #{session} -x #{session_label}" do |cmd|
+      cmd.wait
+      assert_equal session_output, cmd.output.chomp
+    end
+
+    # fail to retreive session login from default session:
+    run_command "kbsecret login -x #{session_label}" do |cmd|
+      cmd.wait
+      assert_match error_pattern_no_record, cmd.stderr
+    end
+
+    # fail to retreive default login from session:
+    run_command "kbsecret login -x -s #{session} #{default_label}" do |cmd|
+      cmd.wait
+      assert_match error_pattern_no_record, cmd.stderr
+    end
+
+    # remove session:
+    run_command_and_stop "kbsecret session rm -d #{session}"
+
+    # fail to retreive login from removed session:
+    run_command "kbsecret login -s #{session} #{session_label}" do |cmd|
+      cmd.wait
+      assert_match error_pattern_no_session, cmd.stderr
+    end
+  ensure
+    # remove default login:
+    run_command_and_stop "kbsecret rm #{default_label}"
+
+    # fail to retreive default login:
+    run_command "kbsecret login -x #{default_label}" do |cmd|
```

Comment:
This should probably be somewhere other than the `ensure` block. It looks like we've covered this part of the code pretty well, though, so it'd be okay to drop this test if you don't want to fit it somewhere else.